compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ConvertConv2DToWinograd.cpp

```diff
@@ -291,28 +165,60 @@ class ConvertConvToWinograd final : public OpRewritePattern<ConvOp> {
   LogicalResult matchAndRewrite(ConvOp convOp,
                                 PatternRewriter &rewriter) const override {
 
-    bool isNchw;
-    if (!isValidConv2d(convOp, isNchw)) {
+    bool isNchwFchw;
+    if (!isValidConv2d(convOp, isNchwFchw)) {
       return failure();
     }
 
-    // Check that kernel has been constant folded (by validating rank = 3)
+    // Create winograd filter transform op.
     Value kernel = convOp.getInputs()[1];
     auto kernelType = cast<ShapedType>(kernel.getType());
     if (!kernelType) {
       return failure();
     }
-    Type elementType = kernelType.getElementType();
-    ArrayRef<int64_t> kernelShape = kernelType.getShape();
-    if (kernelShape.size() != 3) {
-      return failure();
+    if (!kernelType.hasStaticShape()) {
+      return rewriter.notifyMatchFailure(convOp->getLoc(),
+                                         "Kernel shape is not static");
     }
+    SmallVector<int64_t> kernelShape(kernelType.getShape());
+    const int64_t kh = isNchwFchw ? kernelShape[2] : kernelShape[0];
+    const int64_t kw = isNchwFchw ? kernelShape[3] : kernelShape[1];
+    if ((kh != 3) || (kw != 3)) {
```

Comment:
nit: the braces are not needed

```suggestion
    if (kh != 3 || kw != 3) {
```